internal/pkg/cli/storage_init.go

```diff
@@ -1072,13 +1112,93 @@ For example, in JavaScript you can write:
 
 	deployCmd := fmt.Sprintf("copilot deploy --name %s", o.workloadName)
 	actionDeploy := fmt.Sprintf("Run %s to deploy your storage resources.", color.HighlightCode(deployCmd))
-	logRecommendedActions([]string{
+	return []string{
 		actionRetrieveEnvVar,
 		actionDeploy,
-	})
+	}
+}
+
+func (o *initStorageOpts) actionsForEnvStorage() []string {
+	envDeployAction := fmt.Sprintf("Run %s to deploy your environment storage resources.", color.HighlightCode("copilot env deploy"))
+	svcMftAction := fmt.Sprintf("Update the manifest for your %q workload:\n%s", o.workloadName, color.HighlightCodeBlock(o.manifestSuggestion()))
+	svcDeployAction := fmt.Sprintf("Run %s to deploy the workload so that %s has access to %s storage.",
+		color.HighlightCode(fmt.Sprintf("copilot svc deploy --name %s", o.workloadName)),
+		color.HighlightUserInput(o.workloadName),
+		color.HighlightUserInput(o.storageName))
+	addIngressAction := fmt.Sprintf("From the workspace where %s is, run:\n%s",
+		color.HighlightUserInput(o.workloadName),
+		color.HighlightCodeBlock(o.addIngressSuggestion()))
+	if envAndWkldWritten := o.addIngressFrom == "" && o.workloadExists; envAndWkldWritten {
+		return []string{envDeployAction, svcMftAction, svcDeployAction}
+	}
+	if onlyEnv := !o.workloadExists; onlyEnv {
+		return []string{envDeployAction, addIngressAction}
+	}
+	if onlyWkld := o.addIngressFrom != ""; onlyWkld {
+		return []string{svcMftAction, svcDeployAction}
+	}
 	return nil
 }
 
+func (o *initStorageOpts) manifestSuggestion() string {
+	logicalIDSafeStorageName := template.StripNonAlphaNumFunc(o.storageName)
+	switch {
+	case o.storageType == s3StorageType:
+		return fmt.Sprintf(`variables:
+  DB_NAME:
+    from_cfn: ${COPILOT_APPLICATION_NAME}-${COPILOT_ENVIRONMENT_NAME}-%sBucket`, logicalIDSafeStorageName)
+	case o.storageType == dynamoDBStorageType:
+		return fmt.Sprintf(`variables:
+  DB_NAME:
+    from_cfn: ${COPILOT_APPLICATION_NAME}-${COPILOT_ENVIRONMENT_NAME}-%sTableName`, logicalIDSafeStorageName)
+	case o.storageType == rdsStorageType && o.workloadType == manifestinfo.RequestDrivenWebServiceType:
+		return fmt.Sprintf(`secrets:
+  DB_SECRET:
+    from_cfn: ${COPILOT_APPLICATION_NAME}-${COPILOT_ENVIRONMENT_NAME}-%sAuroraSecret`, logicalIDSafeStorageName)
+	case o.storageType == rdsStorageType && o.workloadType != manifestinfo.RequestDrivenWebServiceType:
+		return fmt.Sprintf(`network:
+  vpc:
+    security_groups:
+      - from_cfn: ${COPILOT_APPLICATION_NAME}-${COPILOT_ENVIRONMENT_NAME}-%sSecurityGroup
+secrets:
+  DB_SECRET:
+    from_cfn: ${COPILOT_APPLICATION_NAME}-${COPILOT_ENVIRONMENT_NAME}-%sAuroraSecret`,
+			logicalIDSafeStorageName, logicalIDSafeStorageName)
+
+	}
+	return ""
+}
+
+func (o *initStorageOpts) addIngressSuggestion() string {
+	return fmt.Sprintf(`copilot storage init -n %s \
+--storage-type %s \
```

Comment:
Just to double check my understnading, we are not writing `--lifecycle environment` here on purpose because this addon will be a workload addon that grants access to the workload ecs tasks to the environment level s3 bucket right?

Reply:
> this addon will be a workload addon that grants access to the workload ecs tasks to the environment level s3

The addon produced by `--add-ingress-from` is indeed always a workload-level addon that grants the workload access to the environment storage (s3/ddb/rds)

However, the reasons why we are not writing `--lifecycle` are not only that. `--lifecycle` specifies the storage's lifecycle, not any addon's lifecycle. Since `--add-ingress-from` is not a storage addon, but an ingress addon, `--lifecycle` does not apply. 

> Since `--lifecycle` is useless when using `--add-ingress-from`, then why we don't error out if `--add-ingress-from` and `--lifecycle` are specified together? 

We totally could. In fact, we error out for `copilot storage init --add-ingress-from xx --lifecycle workload` - Copilot does not support adding additional ingress to a workload storage because of best practice. 

The only "compatible" value for `--lifecycle` is "environment" in this case, because `--add-ingress-from` is used to add an ingress to an "environment-level storage".